lesson16/main.js

```diff
@@ -0,0 +1,125 @@
+   loadingIcon.src = "img/loading-circle.gif";
+   body.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+createTabContents();
+async function createTabContents() {
+   startLoading();
+   let responseData;
+   try {
+      responseData = await fetchData(jsonURL);
+      console.log(responseData);
+      if(responseData.length === 0) {
+         displayMessage("データはまだありません");
+         return;
+         } 
+   } catch (error) {
+      displayMessage(error);
+      console.log(error);
+   } finally {
+      endLoading();
+   }
+   createCategoryTab(responseData);
+   renderTabContainer(responseData);
+}
+
+const createCategoryTab = (values) => {
+   const fragment = document.createDocumentFragment();
+   for (let i = 0; i < values.length; i++) {
+      const categoryTabItem = createElementWithClassName("li", "category-tab__item")
+      categoryTabItem.id = `js-categoryTabItem${i}`;
+      const categoryTabText = createElementWithClassName("a", "category-tab__text");
+      categoryTabText.id = `js-categoryTabText${i}`
+      categoryTabText.textContent = values[i].category
+      fragment.appendChild(categoryTabItem).appendChild(categoryTabText);
+   }
+   categoryTab.appendChild(fragment);
+   //初期表示のタブの選択はデータとして持っているがまだ理解できていないので、現段階ではNewsのタブとしてある
+   const newsTabItem = document.getElementById("js-categoryTabItem0");
+   newsTabItem.classList.add("is-active");
+}
+
+const renderTabContainer = (values) => {
+   const tabFragment = document.createDocumentFragment();
+   const tabContainer = createElementWithClassName("div", "tab-container");
+   tabContainer.id = "js-tabContainer";
+   const articlesData = values.map((value) => value.articles);
+   for (let i = 0; i < values.length; i++) {
+      const tabContents = createElementWithClassName("div", "tab-contents");
+      tabContents.id = `js-tabContents${i}`;
+      const articlesTitles = articlesData[i].map((value) => value.title);
+      const articlesHref = articlesData[i].map((value) => value.href);
+      const tabLists = createElementWithClassName("ul", "tab-contents__lists");
+      tabLists.id = `js-tabLists${i}`;
+      const tabImage = createElementWithClassName("img", "tab-contents__image");
+      tabImage.id = `js-tabImage${i}`;
+      tabImage.src = values[i].img;
+      console.log(values);
+      console.log(articlesTitles);
+      tabContents.appendChild(tabImage);
+      for (let j = 0; j < articlesTitles.length; j++) {
+         const tabItems = createElementWithClassName("li", "lists__item");
+         const tabItemsHref = createElementWithClassName("a", "lists__item__text");
+         tabItemsHref.href = articlesHref[j];
+         tabItemsHref.textContent = articlesTitles[j];
+         tabFragment.appendChild(tabContainer).appendChild(tabContents).appendChild(tabLists).appendChild(tabItems).appendChild(tabItemsHref);
+      }
+   }
+   body.appendChild(tabFragment);
```

Comment:
The script tag was placed between the tab and the content, which I felt was not appropriate.
<a href="https://gyazo.com/d08a1bbd39f2e6f0d48a7fcaee6b044a"><img src="https://i.gyazo.com/d08a1bbd39f2e6f0d48a7fcaee6b044a.png" alt="Image from Gyazo" width="474"/></a>

Reply:
I hadn't been able to check on that part.
Thanks for pointing out that.
I changed the insertion point.
[8c5d39d](https://github.com/taka0209/js_lesson/pull/18/commits/8c5d39d70d5b6544e84252726240c837c66d785a)